geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala

```diff
@@ -112,7 +112,26 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     val qp = planQuery(filter, iteratorConfig.iterator, output, keyPlanner, cfPlanner)
 
     val table = acc.getSpatioTemporalTable(sft)
-    val iterators = qp.iterators ++ List(Some(stiiIterCfg), densityIterCfg).flatten
+    var iterators: Seq[IteratorSetting] = qp.iterators ++ List(Some(stiiIterCfg), densityIterCfg).flatten
+
+    if(query.getHints.containsKey(QUERY_SIZE_KEY)) {
+      val cfg = new IteratorSetting(iteratorPriority_SimpleFeatureFilteringIterator, "query-size-iterator", classOf[QuerySizeIterator])
+      //cfg.addOption(QuerySizeIterator.ORIGINAL_SFT_OPTION,sft) // JNH: This requires the original ECQL filter.
+
+      val combinedFilter = (ofilter, ecql) match {
+        case (Some(st), Some(ecql)) => filterListAsAnd(Seq(st, ecql))
+        case (Some(_), None)        => ofilter
+        case (None, Some(_))        => ecql
+        case (None, None)           => None
+      }
+      configureFeatureType(cfg, sft)
+      configureFeatureEncoding(cfg, featureEncoding)
+      configureTransforms(cfg, query)  // Do we need transforms?
+      configureEcqlFilter(cfg, combinedFilter.map(ECQL.toCQL))
+
+      iterators = Seq(cfg)
+    }
```

Comment:
(style) I'd extract this entire function to make the query planner more readable.